### PR TITLE
HWY-207: Fix the code to get rid of warnings about too late timestamps.

### PR DESCRIPTION
**Pull request by goral09**
https://casperlabs.atlassian.net/browse/HWY-207

The warnings were printed for cases that were already covered with different tests (like "is our own proposal?", "is the creator faulty?", etc.) but those checks were done later than the test for `earliest_unit_timestamp`. The solution was simply to rearrange the code so that the `earliest_unit_timestamp is greater than …` is logged only if that's the last error possible.

There was also another small issue in the DES tests setup where it used `Timestamp::now` which was different by a millisecond or two between validators.

I've also modified how equivocation creation is done in the tests to make it more successful to equivocate rather than create an invalid message.

### CODE REVIEW

File: node/src/components/consensus/highway_core/highway_testing.rs [hunk 1]
```diff
@@ -244,7 +244,10 @@ impl HighwayValidator {
                                 // TODO: Don't send both messages to every peer. Add different
                                 // strategies.
                                 let mut wunit = swunit.wire_unit.clone();
-                                wunit.timestamp += 1.into();
+                                match wunit.value.as_mut() {
+                                    None => wunit.timestamp += 1.into(),
+                                    Some(v) => v.push(0),
+                                }
```
Comment (afck) [inline node/src/components/consensus/highway_core/highway_testing.rs:250]: I could swear we already did exactly that. :sweat_smile: (Maybe in a PR that was closed?)

**Review by afck — APPROVED** [review]

File: node/src/components/consensus/highway_core/highway_testing.rs (hunk 1 above)
Comment (goral09) [reply node/src/components/consensus/highway_core/highway_testing.rs:250]: Yes, it was closed.

**Comment by goral09** [comment]
@afck you asked whether the tests get faster after these fixes/changes – looks like they do.

**Comment by goral09** [comment]
bors merge

**Comment by goral09** [comment]
bors cancel

**Comment by goral09** [comment]
bors merge

**Pull request merged**